distribution/sc-distribution-rs/src/lib.rs

```diff
@@ -265,21 +282,26 @@ pub trait Distribution: asset::AssetModule + global_op::GlobalOperationModule {
 
         let mut to_remove_keys = Vec::new();
         let search_gas_limit = self.blockchain().get_gas_left() / 2;
-        for (user_asset_index, user_asset_key) in self.user_locked_asset_map().keys().enumerate() {
-            if (user_asset_index + 1) % GAS_CHECK_FREQUENCY == 0
-                && self.blockchain().get_gas_left() < search_gas_limit
-            {
+        for user_asset_key in self.user_locked_asset_map().keys() {
+            if self.blockchain().get_gas_left() < search_gas_limit {
                 break;
             }
+
             if lower <= user_asset_key.spread_epoch && user_asset_key.spread_epoch <= higher {
                 to_remove_keys.push(user_asset_key);
             }
         }
 
+        let mut removed = 0;
```

Comment:
Instead of keeping an amount of removed, might be better to save the length before entering the "for" and returning old_length - current_length

Reply:
Done